src/parser.ts

```diff
@@ -139,6 +157,18 @@ class Parser {
             const defaultProps = this.extractDefaultPropsFromComponent(exp, source);
             const props = this.getPropsInfo(propsType, defaultProps);
 
+            if (typeof this.opts.propFilter === 'function') {
```

Comment:
I still think it would make sense to change `getPropsInfo` to return `Prop` type which is exactly what you have in `PropItem & PropItemType`. Then use that array with `filter` method.

Reply:
And you're still right! 👍 I've updated the PR.
I re-used the `PropItem` interface to keep API compatibility as `PropItem` is also exported.